### ISSUE TITLE
Logging shows progress bar at each update

### ISSUE THREAD

**Issue by timothybrooks** [wandb/wandb#1265]
The Logs page under a W&B run shows a separate line for each update of a tqdm progress bar. This makes it challenging to find desired logs in the mix, since the number of updates to a progress bar is often equal to the number of training batches, which can reach millions. Since progress bars are commonly used in deep learning libraries (e.g. Pytorch Lightning) to track training, it would be great to have tqdm progress bars logged as they are displayed on the command line or to suggest a workaround to filter out or disable logging of progress bar updates. 

<img width="1440" alt="Screen Shot 2020-09-23 at 5 19 55 PM" src="https://user-images.githubusercontent.com/10535711/94086849-01dd0980-fdc1-11ea-9755-52394800585a.png">

**Comment by vanpelt** [comment]
@timothybrooks thanks for reporting, we're working on a fix ASAP.

**Comment by alexmirrington** [comment]
As a side note, the same occurs for lines printed to `stdout` that end in a return character instead of a newline character, _e.g._ `print(f"loss: {loss.item()}", end="\r")`. I am mainly reporting this because as far as I am aware, `tqdm` prints to `stderr`.

**Comment by yangj1e** [comment]
Any fix to this? I got the same problem.

**Comment by Haoxiang-Wang** [comment]
Got the same problem when using pytorch-lightning & wandB. Strangely, in the first epoch, the progress bar works fine. However, after the first epoch, the bar gets messy. Don't know what's happening inside.

**Comment by vanpelt** [comment]
@Haoxiang-Wang @yangj1e  you can set the environment variable `WANDB_CONSOLE=off` which is a temporary fix.  The next release of the library should fix this issue for good.

**Comment by karthi0804** [comment]
Is this fixed ?

**Comment by karthi0804** [comment]
> @Haoxiang-Wang @yangj1e you can set the environment variable `WANDB_CONSOLE=off` which is a temporary fix. The next release of the library should fix this issue for good.

@vanpelt  This completes disables the log. Is this fix in the latest release? Please let me know any other solution.

**Comment by vanpelt** [comment]
@karthi0804 I had thought we fixed this with the most recent release?  Are you still seeing this?  I'm tagging the engineer on our team @farizrahman4u that's working on console syncing issues.

**Comment by farizrahman4u** [comment]
Tested just now on [Windows](https://wandb.ai/farizrahman4u/uncategorized/runs/s5eyvpsa), [Linux](https://wandb.ai/farizrahman4u/uncategorized/runs/3emjf5gu) and [Colab](https://wandb.ai/farizrahman4u/uncategorized/runs/2iviif2d). The logs are as expected. This is the test script used:

```python

# !pip install wandb

import wandb
import tqdm
import time


run = wandb.init()

for i in tqdm.tqdm(range(100)):
    run.log({"acc": i / 100})
    time.sleep(0.1)

run.finish()
````

If you run into any edge cases let us know. Closing this for now.

**Comment by karthi0804** [comment]
> @karthi0804 I had thought we fixed this with the most recent release? Are you still seeing this? I'm tagging the engineer on our team @farizrahman4u that's working on console syncing issues.

I checked after upgrading. It is fixed in the latest version 0.10.28. Thanks.

**Comment by skrbnv** [comment]
tqdm logging still a bit buggy:

![tqdm](https://user-images.githubusercontent.com/8449626/149722455-bf433f0f-d0f0-4536-8cbf-072545a869be.png)

**Comment by vwrj** [comment]
Hi @skrbnv. Thanks for letting us know - we'll try to reproduce this. Can you send us:
- the link to your wandb.ai workspace -- specifically, the run that produced this log?
- your script or a code snippet which covers logging during your train epoch. 

Are you printing anything to stdout i.e `print()` statements? Or is this purely tqdm logging?

**Comment by ramit-wandb** [comment]
Hi @skrbnv,

I wanted to follow up here on this issue. Are you still experiencing this error with TQDM? If you are, we would appreciate a code snippet reproducing this issue so that we can resolve this on our end.

Thanks,
Ramit
Weights and Biases Support

**Comment by skrbnv** [comment]
Hi @ramit-wandb 
not sure how to reproduce this issue TBH :) It happens very rarely, when I'm AFK, and I still can't figure out circumstances.
Here's a link for the run when it appeared once again lately:
https://wandb.ai/my3bikaht/triplets-voxceleb2-fastrun/runs/3jz7ggje/logs?workspace=user-my3bikaht
As you'll see everything was fine until epoch 31 here
PS. There's no output to console just a basic training/validation loop.

**Comment by dhruva77** [comment]
It is also bugging out for me. Link https://wandb.ai/dhruva77/huggingface/runs/15pgts2i/logs?workspace=user-dhruva77
It bugs after around 20000 iterations. 
Code snippet for reproducibility (I am using the integration with HuggingFace):
```
from transformers import AutoTokenizer, AutoModelForQuestionAnswering, AutoModelForSequenceClassification
from transformers import TrainingArguments, Trainer
from datasets import load_metric

def compute_metrics(eval_pred):
    metric = load_metric("accuracy")
    logits, labels = eval_pred
    predictions = np.argmax(logits, axis=-1)
    return metric.compute(predictions=predictions, references=labels)

    
training_args = TrainingArguments(
            output_dir=output_dirpath,
            learning_rate=3e-6,
            per_device_train_batch_size=6,
            per_device_eval_batch_size=6,
            num_train_epochs=5,
            weight_decay=0.01,
            adam_epsilon=1e-6, #not sure maybe decrease to 1e-8
            evaluation_strategy="steps",
            eval_steps=2000,
            save_strategy="steps",
            save_steps=2000,
            fp16=True,
            save_total_limit = 5, # Only last 5 models are saved. Older ones are deleted.
            load_best_model_at_end=True,
        )


tokenizer = AutoTokenizer.from_pretrained(model_identifier, model_max_length=max_model_length)

model = AutoModelForSequenceClassification.from_pretrained(
            model_identifier, num_labels=2).to(device)




trainer = Trainer(
    model=model,
    args=training_args,
    train_dataset=small_train_dataset,
    eval_dataset=small_eval_dataset,
    compute_metrics=compute_metrics,
)

trainer.train()
```

**Comment by TonyLianLong** [comment]
I also found that the downloaded log files omit some lines with the above `tqdm` effect. However, these lines are present in the web view of the log tab, which means they are captured and uploaded.

**Comment by Ken4scholars** [comment]
I also have an issue with the progress bar with tqdm. My code runs multiple pipelines sequentially each having its own progress bar. Most times, after one pipeline is done, and the next pipeline starts, wandb starts a new progress bar just like in my local console but then immediately goes back to keep updating the older progress bar instead. It's quite a weird behaviour and I was looking for how to configure the console logging frequency as it could be connected to that but could not find such configuration. 
Below is an example of the console logs on the web UI:
```
 COMPUTING FOR GROUP S4
 32%|███▏      | 228/715 [03:43<04:21,  1.86it/s]
{'TP': 20, 'FP': 0, 'FN': 0, 'TN': 6, 'Total': 26, 'Precision': 1.0, 'Recall': 1.0, 'TPR': 1.0, 'FPR': 0.0}
TRANSITION MATRIX COMPUTATION TIME = 1 seconds
 COMPUTING FOR GROUP S5_diff
{'TP': 2724, 'FP': 0, 'FN': 0, 'TN': 312, 'Total': 3036, 'Precision': 1.0, 'Recall': 1.0, 'TPR': 1.0, 'FPR': 0.0}
TRANSITION MATRIX COMPUTATION TIME = 253 seconds
  0%|          | 0/364 [00:00<?, ?it/s]
```
After group `S4` is computed and group `S5_diff` is started, it jumps back and starts updating the group `S4` progress bar

**Comment by shacharp** [comment]
I'm also having issues with tqdm logging (see the attached image):
1 – in one epoch I see the 100% bar, but on the other – 0% (both epochs went ok regarding training).
2 – in some point, the 1 tqdm line turns into many lines, and it keeps that way (taking a lot of space which is annoying)

![image](https://user-images.githubusercontent.com/49400601/216806986-b03d8df9-1cbf-489a-a96b-599dc28e59ff.png)

**Comment by johnnysclai** [comment]
I just recent encounter this issue. `WANDB_CONSOLE=off` doesn't work for me.

**Comment by tristanengst** [comment]
TQDM-derived logs is still an issue, at least with offline SLURM runs.

**Comment by elisim** [comment]
Still happening in the latest release. This issue is super important because wandb logs become useless @ramit-wandb 

<img width="699" alt="image" src="https://github.com/wandb/wandb/assets/17675462/0033d9de-3a97-438f-bd0a-af943bb61080">

**Comment by RoyalBosS-Ayush** [comment]
Same happening with me as well

**Comment by kptkin** [comment]
release 0.18.0 should fix this, closing the issue if there still a problem please re-open

**Comment by henrytsui000** [comment]
Hi @kptkin,

I use torch lightning's RichProgressBar, and it shows as follow:

<img width="1160" height="358" alt="Image" src="https://github.com/user-attachments/assets/c6cab3ad-0d9c-471c-986b-479549901775" />

**Comment by tinker495** [comment]
@kptkin 
Same here! I'm using custom rich live progress bar.
<img width="788" height="495" alt="Image" src="https://github.com/user-attachments/assets/0e6930cc-29c5-44bd-95e9-4f6b84332aec" />
